packages/app-page-builder/src/admin/plugins/pageDetails/header/pageOptionsMenu/PageOptionsMenu.tsx

```diff
@@ -44,9 +48,21 @@ const PageOptionsMenu = props => {
         )
     });
 
+    const { showConfigureDomainDialog } = useConfigureDomainDialog(getDomain(), refreshSiteStatus);
+
     // We must prevent opening in new tab - Cypress doesn't work with new tabs.
     const target = window.Cypress ? "_self" : "_blank";
 
+    const handlePreviewClick = () => {
```

Comment:
Let's wrap this with the `useCallback`. You can use the `[page.id]` for the second arg.